holidays/countries/kenya.py

```diff
@@ -10,65 +10,134 @@
+
+        # Labor Day.
+        dts_observed.add(self._add_labor_day(tr("Labour Day")))
+
+        if self._year >= 2011:
+            # Madaraka Day.
+            dts_observed.add(self._add_holiday_jun_1(tr("Madaraka Day")))
+
+        if 1990 <= self._year <= 2009 or self._year >= 2018:
+            if self._year >= 2025:
+                # Mazingira Day.
+                name = tr("Mazingira Day")
+            elif self._year >= 2021:
+                # Utamaduni Day.
+                name = tr("Utamaduni Day")
+            else:
+                # Moi Day.
+                name = tr("Moi Day")
+            dts_observed.add(self._add_holiday_oct_10(name))
+
+        dts_observed.add(
+            self._add_holiday_oct_20(
+                # Mashujaa Day.
+                tr("Mashujaa Day")
+                if self._year >= 2011
+                # Kenyatta Day.
+                else tr("Kenyatta Day")
+            )
+        )
 
-        # Good Friday
-        self._add_good_friday("Good Friday")
+        dts_observed.add(
+            self._add_holiday_dec_12(
+                # Jamhuri Day.
+                tr("Jamhuri Day")
+                if self._year >= 2011
+                # Independence Day.
+                else tr("Independence Day")
+            )
+        )
 
-        # Easter Monday
-        self._add_easter_monday("Easter Monday")
+        # Christmas Day.
+        dts_observed.add(self._add_christmas_day(tr("Christmas Day")))
 
-        # Labour Day
-        self._add_observed(self._add_labor_day("Labour Day"))
+        # Boxing Day.
+        dts_observed.add(self._add_christmas_day_two(tr("Boxing Day")))
 
-        if self._year >= 2010:
-            # Mandaraka Day
-            self._add_observed(self._add_holiday_jun_1("Madaraka Day"))
+        # Eid-al-Fitr.
+        dts_observed.update(self._add_eid_al_fitr_day(tr("Idd-ul-Fitr")))
 
-        if 2002 <= self._year <= 2009 or self._year >= 2018:
-            self._add_observed(
-                # Utamaduni/Moi Day
-                self._add_holiday_oct_10("Utamaduni Day" if self._year >= 2021 else "Moi Day")
-            )
+        if self.observed:
+            self._populate_observed(dts_observed)
 
-        self._add_observed(
-            # Mashuja/Kenyatta Day
-            self._add_holiday_oct_20("Mashujaa Day" if self._year >= 2010 else "Kenyatta Day")
-        )
+    def _populate_hindu_holidays(self):
+        # Additional public holidays for all persons belonging to the Hindu faith.
 
-        # Jamhuri Day
-        self._add_observed(self._add_holiday_dec_12("Jamhuri Day"))
+        if self._year >= 1984:
+            # Diwali.
+            self._add_diwali(tr("Diwali"))
 
-        # Christmas Day
-        self._add_observed(self._add_christmas_day("Christmas Day"), rule=SUN_TO_NEXT_TUE)
+    def _populate_islamic_holidays(self):
+        # Additional public holidays for all persons belonging to the Islamic faith.
```

Comment:
```suggestion
        """Additional Islamic public holidays."""
```

Reply:
We don't have to use their wording. Up to you 👍 
But as a docstring it looks more organic 💯